src/components/v2/V2Create/forms/TokenForm/index.tsx

```diff
@@ -166,16 +173,22 @@ export default function TokenForm({
       redemptionRate:
         (canSetRedemptionRate && fundingCycleMetadata?.redemptionRate) ||
         defaultFundingCycleMetadata.redemptionRate,
+      weight: fundingCycleData?.weight ?? DEFAULT_ISSUANCE_RATE.toString(),
     }),
     [
       fundingCycleMetadata.reservedRate,
       fundingCycleMetadata?.redemptionRate,
       canSetDiscountRate,
       fundingCycleData?.discountRate,
+      fundingCycleData?.weight,
       canSetRedemptionRate,
     ],
   )
 
+  useLayoutEffect(() => {
```

Comment:
consider `useEffect` instead

Reply:
This is something we only want to run on initial load though, if we have it constantly updating it fooks up because the state lags a bit. I should've added a comment clarifying this

Reply:
@johnnyd-eth I think useLayoutEffect still runs on re-renders though, no?

Only need to use it in very specific circumstances: https://kentcdodds.com/blog/useeffect-vs-uselayouteffect

Reply:
if we only want it to run on initial render, we can use a `useEffect` with an empty dependencies array.

https://css-tricks.com/run-useeffect-only-once/

Reply:
Ahh nice nice very cool, my misunderstanding. Will do